io/disk/ltp_fs.py

```diff
@@ -0,0 +1,116 @@
+# Author: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>
+#
+# Based on code by Martin Bligh <mbligh@google.com>
+# copyright 2006 Google, Inc.
+# https://github.com/autotest/autotest-client-tests/tree/master/ltp
+
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process, archive
+from avocado.utils.software_manager import SoftwareManager
+
+
+class ltp(Test):
+    '''
+    LTP (Linux Test Project) testsuite
+    '''
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        deps = ['gcc', 'make', 'automake', 'autoconf']
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        self.disk = self.params.get('disk')
+        self.mount_point = self.params.get('mount_point', default=self.srcdir)
+        self.script = self.params.get('script')
+        self.fs = self.params.get('fs', default='ext4')
+        self.args = self.params.get('args', default='').split(',')
+        self.log.info("creating ext4 filesystem on disk %s" % self.disk)
+        create_fs = "mkfs.%s %s" % (self.fs, self.disk)
+        if process.system(create_fs, shell=True, ignore_status=True) != 0:
+            self.fail("Creating filesystem %s on\
+                       %s failed" % (self.fs, self.disk))
+        self.log.info("Mounting disk %s to mount location %s"
+                      % (self.disk, self.mount_point))
+        mounting = "mount %s %s" % (self.disk, self.mount_point)
+        if process.system(mounting, shell=True, ignore_status=True) != 0:
+            self.fail("Unable to mount to location  %s" % self.mount_point)
+        url = "https://github.com/linux-test-project/ltp/\
+               archive/master.zip"
+        tarball = self.fetch_asset("ltp-master.zip",
+                                   locations=[url], expire='7d')
+        archive.extract(tarball, self.srcdir)
+        ltp_dir = os.path.join(self.srcdir, "ltp-master")
+        os.chdir(ltp_dir)
+        build.make(ltp_dir, extra_args='autotools')
+        ltpbin_dir = os.path.join(ltp_dir, 'bin')
+        os.mkdir(ltpbin_dir)
+        process.system('./configure --prefix=%s' % ltpbin_dir)
+        build.make(ltp_dir)
+        build.make(ltp_dir, extra_args='install')
+
+    def test(self):
+        for val in self.args:
+            if self.script == 'runltp':
+                logfile = os.path.join(self.logdir, 'ltp.log')
+                failcmdfile = os.path.join(self.logdir, 'failcmdfile')
+                val += (" -q -p -l %s -C %s -d %s"
+                        % (logfile, failcmdfile, self.mount_point))
+            self.log.info("Args = %s" % self.args)
+            ltpbin_dir = os.path.join(self.srcdir, "ltp-master", 'bin')
+            cmd = os.path.join(ltpbin_dir, self.script) + ' ' + val
```

Comment:
using + for concatenation of string not looks good can you please use like this 
`
 cmd ='%s   %s' % (os.path.join(ltpbin_dir, self.script), val)`